spec/rails_helper.rb

```diff
@@ -19,7 +19,13 @@
   Capybara::Selenium::Driver.new app, browser: :chrome, desired_capabilities: capabilities
 end
 
+Capybara.register_driver :firefox do |app|
+  options = Selenium::WebDriver::Firefox::Options.new
+  Capybara::Selenium::Driver.new(app, :browser => :firefox, :options => options)
+end
+
 Capybara.javascript_driver = :selenium_chrome_headless
+#Capybara.javascript_driver = :firefox
```

Comment:
yes, for future reference. It doesn't pass the whole suite on FF due to a test timing issue. I can't bring myself to add "sleep" or "wait", but it's tricky to fix and I'm not sure of the right solution. But it is worth running to validate any changes you think might have cross-browser issues as evidenced by this very PR.

Reply:
The `wait` in capybara isn't a sleep, it's more of a max retry timeout:
https://github.com/teamcapybara/capybara#asynchronous-javascript-ajax-and-friends

The default wait before it gives up is two seconds.  

If there are test timing problems, no problem adding stuff like the thing I did here:
https://github.com/ophrescue/RescueRails/pull/832/files#diff-b31598bf6bf3a42245764bd258444b32

Reply:
@markottaviani I'll try 'wait' to see if I can make FF testing more reliable. I just discovered that FF now has a headless mode, so we can consider running the full suite on two browsers. I think this would have to be invoked via rake iso rspec.
If this seems like a good idea, please make a new "issue".